coreLib/src/main/java/dev/kosmx/playerAnim/core/data/AnimationFormat.java

```diff
@@ -11,8 +14,33 @@ public enum AnimationFormat {
     SERVER(null),
     UNKNOWN(null);
 
-    private final String extension;
+    private static final Map<String, AnimationFormat> FORMATS;
+
+    static {
+        AnimationFormat[] formatsValues = values();
+
+        FORMATS = new HashMap<>(formatsValues.length);
+
+        for (AnimationFormat format : formatsValues) {
+            if (format.extension != null) 
+                FORMATS.putIfAbsent(format.extension, format);
+        }
+    }
 
+    public static AnimationFormat byFileName(String fileName) {
```

Comment:
This can be used in the server side of EmoteCraft where it loads emotes from a folder (reduces load time), I'll send a pullrequest later on

Reply:
If it is done like that, please allow custom formats to be added

Reply:
For what?